spec/facility_locations_spec.rb

```diff
@@ -0,0 +1,110 @@
+    end
+  end
+
+  describe 'Creating Objects from Multiple Data Sets & Analyzing Data' do
+    it 'can #create_facilities from external data sources' do
+      colorado = DmvDataService.new.co_dmv_office_locations
+      co_offices = FacilityLocations.new
+      co_offices.create_facilities(colorado)
+
+      expect(co_offices.facilities.length).to be > 1
+      co_offices.facilities.each do |facility|
+        expect(facility).to be_an_instance_of(Facility)
+        expect(facility.name.nil?).to eq(false)
+        expect(facility.name).to be_an_instance_of(String)
+        expect(facility.address.nil?).to eq(false)
+        expect(facility.address).to be_an_instance_of(Array)
+        expect(facility.address[-2][:state]).to eq("CO")
+        expect(facility.phone.nil?).to eq(false)
+        expect(facility.phone).to be_an_instance_of(String)
+      end
+    end
+
+    it 'can #create_facilities from multiple data sources' do
+      new_york_facilities = DmvDataService.new.ny_dmv_office_locations
+      ny_offices = FacilityLocations.new
+      ny_offices.create_facilities(new_york_facilities)
+  
+      ny_offices.facilities.each do |facility|
+        expect(facility).to be_an_instance_of(Facility)
+        expect(facility.name.nil?).to eq(false)
+        expect(facility.name).to be_an_instance_of(String)
+        expect(facility.address.nil?).to eq(false)
+        expect(facility.address).to be_an_instance_of(Array)
+        expect(facility.address[-2][:state]).to eq("NY")
+        #Some offices in NY have numbers, some don't
+      end
+
+      missouri_facilities = DmvDataService.new.mo_dmv_office_locations
+      mo_offices = FacilityLocations.new
+      mo_offices.create_facilities(missouri_facilities)
+  
+      mo_offices.facilities.each do |facility|
+        expect(facility).to be_an_instance_of(Facility)
+        expect(facility.name.nil?).to eq(false)
+        expect(facility.name).to be_an_instance_of(String)
+        expect(facility.address.nil?).to eq(false)
+        expect(facility.address).to be_an_instance_of(Array)
+        expect(facility.address[-2][:state]).to eq("MO")
+        #Inconsisten phone number for offices
+      end
+    end
+
+    it 'can iterate through a hash and find the #hours of operation' do
+      colorado = DmvDataService.new.co_dmv_office_locations
+      co_offices = FacilityLocations.new
+      co_offices.create_facilities(colorado)
+
+      co_offices.facilities.each do |facility|
+        expect(facility.hours.nil?).to eq(false)
+        expect(facility.hours).to be_an_instance_of(String)
+      end
+
+      new_york_facilities = DmvDataService.new.ny_dmv_office_locations
+      ny_offices = FacilityLocations.new
+      ny_offices.create_facilities(new_york_facilities)
+  
+      ny_offices.facilities.each do |facility|
+        expect(facility.hours.nil?).to eq(false)
+        expect(facility.hours).to be_an_instance_of(Hash)
+      end
```

Comment:
I'm not sure exactly what method you're testing here. You can do unit tests for the `hours` method in the `facility_locations` class and test just that method itself. This one is probably more of an integration test which is fine, but for a method like `hours` that is doing quite a bit, it can be helpful to have a unit test for just that method.